frontend/src/libraries/explorer-wamp/blocks.ts

```diff
@@ -61,6 +61,13 @@ export default class BlocksApi extends ExplorerApi {
           paginationIndexer,
         },
       ]);
+      const finalHeight = await this.getFinalStats();
+      if (blocks.length > 0) {
+        blocks.map(
+          (block: any) => (block.isFinal = block.height <= finalHeight)
```

Comment:
Please enlighten me given my nonexistent JS skills. This code looks quite strange to me:
* what is the condition `block.length > 0` for?
* It seems to me that an imperative style would be better here, but please let me if this is some JS pattern.

cc @frol

Reply:
1. if wamp could not fetch, blocks will be null and length is 0
2. we use this pattern in transactions.ts, too.

Reply:
1. `block.length > 0` is not required since an empty array is fine to be mapped.
2. `map` creates a new array of undefined in this case, and we don't need that. Instead, a more readable version would be a simple `for` loop, I agree with @bowenwang1996. Alternatively we can Use `forEach` instead of `map`, but I rarely find it more readable than a simple loop for mutable data.